### ISSUE TITLE
Perian + OS X Mavericks

### ISSUE THREAD

**Issue by Jeezy4** [MaddTheSane/perian#3]
Does that the support will be updated to OS X Mavericks?
Thanks.

**Comment by MaddTheSane** [comment]
Assuming that Mavericks still loads QuickTime codec plug-ins, yes. If there's a replacement, it might take awhile. If there isn't a replacement, then no, sorry.

**Comment by mcandre** [comment]
+1

**Comment by laranea** [comment]
+1

**Comment by andrebalza** [comment]
+1

**Comment by paraita** [comment]
+1

**Comment by anjunatl** [comment]
+1, really missing the Finder thumbnails for non-Apple codecs

**Comment by craigkeller** [comment]
looks like Mavericks uses AVKit instead of QTKit :( No non-apple codec Quicklook anymore.

**Comment by MaddTheSane** [comment]
It is possible to make a QuickLook plug-in that handles non-QuickTime and non-AVFoundation movies. It will need to be written, but I believe it is possible.

**Comment by craigkeller** [comment]
I will dedicate my life in the honor of whomever writes said plugin.

**Comment by applenoyon** [comment]
+1 thanks ;)

**Comment by danieldietrich** [comment]
+1

**Comment by jesstech** [comment]
> It is possible to make a QuickLook plug-in that handles non-QuickTime and non-AVFoundation movies. It will need to be written, but I believe it is possible.

Are you planning to author such a plugin, or can you outline how one might work?

**Comment by MaddTheSane** [comment]
Further research on an audio-only QuickLook plug-in has lead me to believe that  there _are_ APIs for doing audio, they're just private. Someone did make a video-only QuickLook Plug-in, so that, at least, is possible with public APIs.

**Comment by NeoNapster** [comment]
Does Mavericks not support DTS passthrough?

**Comment by MaddTheSane** [comment]
To my knowledge, OS X never had DTS pass through, except maybe the DVD Player and VLC.

**Comment by NeoNapster** [comment]
No, I mean since Mavericks came out the ACS passthrough (Sorry not DTS my bad) is no longer working, on my AV receiver the DTS logo does not show and the sound, sounds muffled........ I've installed the latest version of Perian.. 1.3.1

**Comment by MaddTheSane** [comment]
Perian 1.3.2 has been released. It does introduce a fix for the ac3 pass-through, hopefully.

**Comment by andrebalza** [comment]
Appreciate your effort. is there any ready to download executable?

**Comment by dreshfield** [comment]
@andrebalza: You can download the zipped prefPane [here](https://github.com/MaddTheSane/perian/releases).

**Comment by riddl** [comment]
+1

**Comment by XenitXTD** [comment]
Hey Guys, I installed Version 1.3.2 on mavericks 10.9.2 and all but i don't see it work, I immediately tested the preview(Spacebar) feature like in the past and its not working on the two files i tested (Namely MKV and MPEG) didn't try testing further... i completely deleted old installs and then rebooted after the install with no luck?

Is there anything i should know about it now working right now or did i miss something?

Lastly, I am so glad that you guys decided to continue development on this as this was an absolutely useful tool for the mac that just made life easier

**Comment by thecomputermedic** [comment]
Quick Look no longer uses Quicktime QTKit framework, now they are using
AVKit framework, which does not allow plug-ins.

Brett Goldberg
505-577-2557
Medic@TheComputerER.com

**Comment by thecomputermedic** [comment]
Keep an eye on MaddTheSane as he has taken up the mantle of the now defunct
Perian.

https://github.com/MaddTheSane/perian

Brett Goldberg
505-577-2557
Medic@TheComputerER.com

**Comment by MaddTheSane** [comment]
There is a Flash video QuickLook plug-in out there, so there is a way to play back movies via QuickLook.

Although that plug-in doesn't support audio.

**Comment by jelias** [comment]
That would be a problem, but it's definitely the right step forward!

**Comment by Thomasmac** [comment]
Downloaded and installed Perian 1.3.2 but don't see/notice any difference to version 1.2.3. The only thing that is different: Perian 1.3.2 extended my context menu "open with" - "Perian - Open in QT Player". It does open QT-Player but not QT-Player 7. It opens the QT Player of Mavericks instead which does not make sense really as this player immediately starts encoding instead of playing the flv or avi-file :-).

Question: "There is a Flash Video QuickLook plug-in out there" (without audio). Where is "out there"? I did not find anything browsing the net.

**Comment by MaddTheSane** [comment]
There was a QuickLook plug-in that played back FLV on the internet. Sadly, it uses QTKit to get a video file, a no-no under Mavericks.

**Comment by mcandre** [comment]
Gotta love it when things are deprecated in favor of... no particular replacement. Like OpenID.

**Comment by ghost** [comment]
@MaddTheSane Do you have any news on Perian being able to play WebM/VP8 files within Safari, on Mavericks?
I have found that you can watch webm videos if you open them directly in a new tab, for example, but they will not play inline in Safari. To the best of my knowledge, QuickTime handles Safari video.

**Comment by MaddTheSane** [comment]
QuickTime handles sites that want a plug-in to view video/audio. However, many sites are now moving to HTML5 video tag, which I would not be surprised if it went straight to AVFoundation.

**Comment by ghost** [comment]
Ok, i'm not 100% certain on that, but i was under the impression that HTML5 audio and video is handled by QuickTime in Safari, but like i say, i'm not sure.

**Comment by asoksevil** [comment]
I just experienced this. It is quite weird that you can actually playback Webm files in a separate tab while embedded or inline videos can´t be displayed. I used to be able to do that on Snow Leopard.

**Comment by ghost** [comment]
When a file is loaded in a separate tab, it is handled by the (old) QuickTime Plugin, which can load Perian with no problem. 
However, inline videos don't use QuickTime anymore, they use AVFoundation which doesn't support any kind of extensions. 

In older versions of OSX (before Mavericks I think), inline html5 video was handled by QuickTime, which is why Perian used to work in browser.

**Comment by asoksevil** [comment]
I see, and I guess this hasn´t changed since Mavericks so even El Cap and Yosemite have the same issue,

**Comment by ghost** [comment]
That is correct. We can hope that one day, Apple will allow extensions/plugins to AVFoundation (unlikely) or that Apple might even just implement WebM video decoding into AVFoundation (also unlikely).

My only real advice is, if you visit a webpage with only a WebM video and no alternative, you should use a different browser for that page.

If you have the Develop menu enabled in Safari (Preferences > Advanced > Enable Develop Menu), then when you find a WebM video, you can click “Develop > Open Page With > Google Chrome” or whichever other browser you prefer. Chrome and Firefox both support WebM video.

**Comment by MarcusJohnson91** [comment]
Or you can set the user agent to iOS (iPad), and it'll force the site to load an mp4 version in Safari.

**Comment by newgloton** [comment]
Perian is an open source QuickTime component that enables Apple Inc.'s QuickTime to play many popular video formats that are not compatible with QuickTime natively on Mac OS X. However, Perian isn't being developed anymore. What a pity. If you are using the latest QuickTime 10 or later, Perian no longer works for you. Besides, Perian has only been compatible with Mac OS X since it was created. 

Why just download a Perian alternative for Mac (El Capitan include) and Windows (Windows 10 included)? Here is one: http://www.faasoft.com/articles/perian-alternative-for-mac-and-windows.html

With it, you can convert all kinds of video and audio files and then play the converted files anywhere without any plug-in.

**Comment by kode54** [comment]
Thanks for the spam, pal!

**Comment by Wowfunhappy** [comment]
I'm many years late, but I (finally) got the Mountain Lion QuickTime to work on Mavericks, so I can use Perian. https://github.com/Wowfunhappy/QuickTime-Fixer/releases
